feat(blog) add a post for the new UC in production (#7688)
jenkins-infra/jenkins.io: content/blog/2024/11/16/2024-11-16-new-update-center.adoc

---
layout: post
title: "New Update Center (updates.jenkins.io) Architecture in Production:\n 18 November 2024"
tags:
- jenkins
- jenkins-infra
- update-center
authors:
- dduportal
- smerle33
opengraph:
  image: /images/post-images/2023/01/12/jenkins-newsletter/infrastructure.png
overview: "The new Update Center implementation in production"
discourse: true
---

== Summary (link:https://en.wikipedia.org/wiki/Wikipedia:Too_long;_didn%27t_read[TL;DR])

> Note: this is a follow up of link:/blog/2024/11/07/update-center-brownouts-6/[the 07 and 08 November 2024 24 hours brownout].

Reviewer: ```suggestion
Note: this is a follow up of link:/blog/2024/11/07/update-center-brownouts-6/[the 07 and 08 November 2024 24-hour brownout].
```

> The service link:https://updates.jenkins.io[https://updates.jenkins.io] will switch its implementation to the new system, Monday 17 November 2024 at 09:00 am UTC.

Reviewer: ```suggestion
The service link:https://updates.jenkins.io[https://updates.jenkins.io] will switch its implementation to the new system, on Monday 17 November 2024 at 09:00 am UTC.
```

All Jenkins users are impacted but should not see any functional change.

⚠️ Please check that your organization respects the advertised DNS TTL or you might be stuck in the brownout longer than expected.

== What is the "Update Center"?

> Jenkins link:https://updates.jenkins.io[Update Center] is a web server at the core of the Jenkins public infrastructure that distributes the plugins, tool installers, and versions index to all Jenkins servers across the world.

Reviewer: ```suggestion
Jenkins link:https://updates.jenkins.io[Update Center] is a web server at the core of the Jenkins public infrastructure that distributes the plugins, tool installers, and version index to all Jenkins servers worldwide.
```

From the installation wizard to regular plugin updates, if you run Jenkins, then you use this service under the hood.

> Today, it serves around 50 Tb of data (outbound bandwidth) each month from a single virtual machine on AWS, which costs around $6,000 per month.

Reviewer: ```suggestion
Today, it serves around 50 TB of data (outbound bandwidth) each month from a single virtual machine on AWS, which costs around $6,000 per month.
```

The Jenkins infrastructure team has worked relentlessly over the past years to implement a new sustainable implementation for this service in order to sustain and improve it.

The new Update Center implementation features a highly available system that redirects user requests to a download mirror close to their location.

Additional information is available in the link:https://github.com/jenkins-infra/helpdesk/issues/2649[GitHub issue].

== What Happens?

After running 6 brownouts (e.g. using the new system in production from 1 to 24 hours),
we are confident we ran out of things to break (©link:https://www.jenkins.io/blog/authors/basil/[Basil Crow]).

We'll simply switch the DNS entry `updates.jenkins.io` to this new service and continue monitoring it.

> In 1 month, if no major issue arise, we'll start decommissioning the old service on the VM which was used for years.

Reviewer: ```suggestion
In 1 month, if no major issue arises, we'll start decommissioning the old service on the VM which was used for years.
```
